apps/cli/lib/sync/rpc.ex

```diff
@@ -0,0 +1,230 @@
+defmodule CLI.Sync.RPC do
+  @moduledoc """
+  Script to set-up a sync with Infura.
+  """
+  require Logger
+
+  @save_block_interval 1000
+  @max_retries 5
+
+  @type ethereumex_client :: module()
+  @type state :: ethereumex_client()
+
+  @provider_url "http://mainnet.infura.io"
+
+  @doc """
+  Sets up database and loads chain information
+  """
+  @spec setup(String.t() | nil) :: {:ok, state} | {:error, any()}
+  def setup(provider_url) do
+    provider_url = if provider_url, do: provider_url, else: @provider_url
+
+    {:ok, _started} = Application.ensure_all_started(:ethereumex)
+
+    state =
+      case URI.parse(provider_url) do
+        %URI{scheme: scheme} when scheme == "http" or scheme == "https" ->
+          Application.put_env(:ethereumex, :url, provider_url)
```

Comment:
I think it's a good practice to match it against the expected return value, this case it's :ok

Reply:
Good call.